neural_net_basics/neural-network.md

```diff
@@ -0,0 +1,137 @@
+
+Using the SoftMax algorithm as the activation function for the final layer of a neural network you can convert the values of each classes output into a percentage value, indicating the certainty of the network at its classification.
+
+It has been proven that a neural work which has two layers and uses non-linear activation functions can be a universal function approximator, that can find any mathematical function.
+
+## Determining the weights and biases (the parameters)
+
+So how do the weights and biases of each neuron get set? 
+
+The weights and Biases could be manually updated until the output of the neural network is correct. However neural networks can consist of millions/billions of parameters, so an automated process is required. 
+
+To be able to automate the process a way of evaluating the accuracy of the neural network is required. To do this a loss function can be used to give a value that indicates how far away the current neural network is from making an accurate evaluation.
+
+### Loss Function
+A common algorithm for determining the error of the neural network is the Mean Squared Error function. This finds a positive value indicating how far from the expected value the network is. 
+
+$$ MSE = \frac{1}{n}\sum_{i=1}^{n}(Y_{i}-Ŷ_{i})^{2} $$ 
+
+* $Ŷ$ is the actual network output
+* $Y$ is the expected value of class
+* $n$ is the total number of output neurons.
+
+The loss function enables an empirical analysis to see if the change in parameters improves or worsens the output of the neural network. The goal is to change the parameters to minimise the loss as much as possible.
+
+A method called Gradient Descent can be used to determine how to adjust the parameters to reduce the loss.
+
+### Gradient Descent
+Assume a single input neural network, with a single weight. If we plotted the weight v error graph we could have something like the following graph.
+
+![png](/neural_net_basics/assets/error_graph.png)
+
+Say the red x is the current value of the error based on the current weight and the green x is the minimum error. 
+
+To calculate the new weight value, take the derivative of the error in respect to the weight and subtract the value from the current weight.
+
+$$ w_{i+1} = w_i - \alpha\frac{\delta L}{\delta w_i}$$
+
+* $w_{i+1}$ is the new weight
+* $w_i$ the current weight
+* $\alpha$ the learning rate
+* $\frac{\delta L}{\delta w_i}$ is the derivative of the change in loss against weight
+
+The learning rate is used to control how much of a change to the weight is made, if the size of the weight change is too big then loss minima may be stepped over or it may never settle in the "valley" of the minima and constantly jump from side to side. If it's too small then it will take longer to reach the local minima.
+
+### Back Propagation - remove?!
```

Comment:
I'd keep it. I think it's fine to be honest and say something along the lines of it's a complicated subject to explain briefly, check out these resources to learn more.